src_blog/_posts/New-Release-Grain-v0.4.0-Cereal.md

```diff
@@ -0,0 +1,67 @@
+
+### Grain format
+
+Grain format provides an opinionated code formatter tool for Grain source code. This tool was an absolutely herculean effort by Marcus, the core team member we introduced in our last release post and maintainer of the Grain Language Server Protocol.
+
+I won’t go in-depth on the formatter in this post, but Marcus wrote up an excellent [blog post](https://grain-lang.org/blog/2021/09/04/grain-formatter/) about it! I recommend you give it a read, and grab our latest vscode extension to have format-on-save; I’ve been using it for a few days and it is a great experience.
+
+### Grain doc
+
+Additionally, I finally finished the initial work around our documentation tool. This provides a JSDoc-like experience for your Grain code. We’ve already begun using it to document the standard library, so we’ll be able to keep our API docs consistently updated.
+
+You can check out the features and syntax in my preview [blog post](https://grain-lang.org/blog/2021/08/27/tooling-preview-graindoc/). I’m particularly happy that the tool infers the types of your parameters and return values, so you don’t have to worry about keeping them in-sync.
+
+## Callee-owned values
+
+This is a bit of an internal change, but it can cause a lot of ripples throughout the ecosystem. We changed the way our compiler outputs Garbage Collector code so that values passed to a function are owned and cleaned up by the function being called. Previously, the caller would `decRef` the values after a function returned; however, we found that this caused some memory leaks in certain scenarios. By changing this memory ownership convention, we now know all parameters are `decRef`ed by the time the function returns.
+
+This only has ramifications to you if you were using the `@disableGC` annotation on your Grain code. While this attribute is not recommended for end-user code, we use it in the stdlib and some library authors might need to use it. If you are in need of this, please get in touch with us to understand the new calling convention.
+
+## Standard library
+
+As with most releases, we had a lot of work happening in the standard library—some by the core team and some by community contributors!
+
+### Number
+
+Thanks to [@spotandjake](https://twitter.com/spotandjake), we finally have a preliminary module for working with our Number type. This module will probably be one of the most used, so we are glad for the contribution. You can find the docs in our [standard library docs](https://grain-lang.org/docs/stdlib/number).
+
+### Sys
+
+Previous to this release, any WASI calls in our `sys/*` modules would throw if they didn’t return a success code. This was a problem for end users because you couldn’t catch the error and handle it, such as when trying to `stat` a file to see if it exists. We’ve changed all of these functions to return a `Result` type instead, so you can react to the errors.
```

Comment:
```suggestion
Prior to this release, any WASI calls in our `sys/*` modules would throw an error if they didn’t return a success code. This was a problem for end users because you couldn’t catch the error and handle it, such as when trying to `stat` a file to see if it exists. We’ve changed all of these functions to return a `Result` type instead, so you can decide how to deal with these errors.
```

Reply:
Combined @peblair's suggestion into mine:
```suggestion
Prior to this release, any system calls via WASI, the WebAssembly System Interface, in our `sys/*` modules would throw an error if they didn’t return a success code. As there isn't yet support for WebAssembly exceptions, this wasn't great because you couldn’t catch the error and handle it, such as when trying to `stat` a file to see if it exists. We’ve changed all of these functions to return a `Result` type instead, so you can now handle any errors that occur.
```